packages/desktop-client/src/components/accounts/Account.js

```diff
@@ -693,11 +693,67 @@ class AccountInternal extends PureComponent {
     return null;
   };
 
-  onReconcile = balance => {
+  lockTransactions = async () => {
+    this.setState({ workingHard: true });
+
+    let { accountId } = this.props;
+
+    let { data } = await runQuery(
+      q('transactions')
+        .filter({ cleared: true, reconciled: false, account: accountId })
+        .select('*')
+        .options({ splits: 'grouped' }),
+    );
+    let transactions = ungroupTransactions(data);
+
+    let changes = { updated: [] };
+
+    transactions.forEach(trans => {
+      let { diff } = updateTransaction(transactions, {
+        ...trans,
+        reconciled: true,
+      });
+
+      transactions = applyChanges(diff, transactions);
+
+      changes.updated = changes.updated
+        ? changes.updated.concat(diff.updated)
+        : diff.updated;
+    });
+
+    await send('transactions-batch-update', changes);
+    await this.refetchTransactions();
+  };
+
+  onReconcile = async balance => {
     this.setState({ reconcileAmount: balance });
   };
 
-  onDoneReconciling = () => {
+  onDoneReconciling = async () => {
+    let { accountId } = this.props;
+    let { reconcileAmount } = this.state;
+
+    let { data } = await runQuery(
+      q('transactions')
+        .filter({ cleared: true, account: accountId })
+        .select('*')
+        .options({ splits: 'grouped' }),
+    );
+    let transactions = ungroupTransactions(data);
+    var cleared = 0;
+
+    transactions.forEach(trans => {
+      if (!trans.is_parent) {
+        cleared += trans.amount;
+      }
+    });
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏
```suggestion
    let cleared = transactions.reduce((sum, trans) => sum + trans.is_parent ? trans.amount : 0, 0);
```